model/labelset.go

```diff
@@ -18,6 +18,8 @@ import (
 	"fmt"
 	"sort"
 	"strings"
+
+	"github.com/mailru/easyjson/jwriter"
```

Comment:
I don't think this is an appropriate dependency to have in this package. Data models should just be data models.

Reply:
@brian-brazil this package already has a dep on [encoding/json](https://github.com/prometheus/common/blob/master/model/labelset.go#L17) meaning marshaling is already a part of the model. Similarly custom marshal and unmarshal methods already exist in this package (and even in [that file](https://github.com/prometheus/common/blob/master/model/labelset.go#L153)).

In golang you can't define methods on a struct outside of the package where the struct is defined, meaning that if you want a struct to adhere to an interface it must define the methods in the same package. We could move the marshal methods into a different file in the same package, but the existing ones are in the same files-- but if separate files in the package is preferred I'd be more than happy to move them around.